harness/determined/keras/_callback.py

```diff
@@ -337,34 +338,8 @@ def _load(self, checkpoint: Optional[str]) -> Optional[contextlib.ExitStack]:
             # Load model.
             self.load_model(self.model, str(path / "model_checkpoint"), self._core.distributed)
 
-            # Load training state also.
```

Comment:
wonder if instead of exitstack this woulda been easier to grok as just .__enter__() and .__exit__() calls

Reply:
I wanted to preserve the natural `__exit__` behavior if an exception happened before the `._load()` exited naturally.

If there was a callback-wide context manager somewhere, I would have installed it there too, as a backup, but I don't think there is.

Reply:
btw I'm happy to go on the record and say that I think exceptions are very sneaky and non-obvious control flows, and I don't think they should exist.

Reply:
yeah that makes sense. you'd have to somehow make sure the `__exit__` ran, seems riskier.